packages/hadron-document/src/element.ts

```diff
@@ -314,19 +360,22 @@ class Element extends EventEmitter {
    *
    * @returns {Element} The new element.
    */
-  insertEnd(key, value) {
+  insertEnd(key: string | number, value: BSONValue): Element {
+    if (!this.elements) {
+      throw new Error('Cannot insert values on non-array/non-object elements');
+    }
     if (this.currentType === 'Array') {
       this.elements.flush();
       key = 0;
       if (this.elements.lastElement) {
         if (this.elements.lastElement.currentKey === '') {
           this.elements.handleEmptyKeys(this.elements.lastElement);
         }
-        key = this.elements.lastElement.currentKey + 1;
+        key = (this.elements.lastElement.currentKey as number) + 1;
       }
     }
-    var newElement = this.elements.insertEnd(key, value, true, this);
-    this._bubbleUp(Events.Adde, newElement);
+    const newElement = this.elements.insertEnd(key, value, true, this);
+    this._bubbleUp(Events.Added, newElement);
```

Comment:
omg 🙈

Reply:
Lol while this is a fix, I'm thinking lets try out the places this `insertEnd` is hit to make sure this bug wasn't being assumed by the callers 🙈